zulipterminal/core.py

```diff
@@ -492,6 +492,21 @@ def show_media_confirmation_popup(
             self, question, callback, location="center"
         )
 
+    def update_search_text_for_narrow(self, search_query: str) -> str:
+        """
+        This function is to check if the user is currently inside a narrow or not
+        and if they are, then the search_query is changed accordingly
+        """
+        # In cases where '+' is already present in stream and/or topic name, it is converted to '%2B'
+        # then the spaces in stream and topic names are converted to '+'
+        if self.model.narrow[0][0] == "stream":
+            search_query += f" {self.model.narrow[0][0]}:{self.model.narrow[0][1].replace('+', '%2B').replace(' ', '+')}"
+        if len(self.model.narrow) > 1:
+            if self.model.narrow[1][0] == "topic":
+                self.model.narrow[1][1].replace(" ", "+")
+                search_query += f" {self.model.narrow[1][0]}:{self.model.narrow[1][1].replace('+', '%2B').replace(' ', '+')}"
+        return search_query
+
     def check_for_prefixes(self, query: str) -> Tuple[str, List[List[str]]]:
```

Comment:
The name while WIP doesn't matter too much, but now we're firming up the logic, let's make this name more expressive. If I see `check_for_prefixes(something)` in code, it doesn't tell me what it does (I can think of various meanings), or why it returns what it does do - I can look it up in an editor or IDE, but better to read right there :)

So, maybe think what goes in the blank:
```python3
search_query, search_narrow = self.<BLANK>(final_search_query)
```

While some/many ZT functions don't have docstrings (or comments) since they're older, some are very simple and ideally well-named that it may be sufficient to understand what's happening when they're used without them.

You might also rename the other function, if you change the logic of what it does.